tests/test_highlighting/test_not_in_list.py

```diff
@@ -0,0 +1,41 @@
+from parameterized import parameterized
+
+from tests.Highlighter import HighlightTester
+
+
+class HighlighterTestList(HighlightTester):
+
+    # @parameterized.expand(HighlightTester.levels(5, 11))
+    def test_text_not_in_list(self):
+        self.assert_highlighted_chr(
+            "if text not in list print 'not in'",
+            "KK TTTT KKK KK TTTT KKKKK SSSSSSSS",
+            level='level5', lang="en", intermediate_tests=False)
```

Comment:
@jpelay I am not sure why, but the only way to run the 'not-in' highlighting tests is through switching off the intermediate_tests. Could you please check why this is the case? Maybe I did sth wrong? Thanks!

Reply:
Hello Boryana! Yes, that's because those tests are meant for Ace 😅 we never ended up removing them ups (opened #5144 so we don't forget),  you can find the Lezer tests in the folder  `cypress/e2e/lezer-tests`. 

Ill explain here a bit how they work:

The tests are composed like this:

```javascript
describe('Lezer tests for level', () => { // This outer one groups the tests for the level
    describe('Successful tests', () => { // this one groups tests of a kind, for example sucessful tests
        describe('print test', () => { // The individual test
            code = 'print hello world' // the code you want to test
            expected = 'Program(Command(Print(print, Text, Text)))` // The tree generated by Lezer
            multiLevelTester('Test print with text', code, expectedTree, minLevel, maxLevel);  // can also be singleLevelTester
        })
    })
})
```

The trick I do to check the tree generated by Lezer in the front-end so I can copy paste it is to add this code in the end of the `setHighlighterForLevel` in the file `cm-editor.ts`:

```javascript
        const transaction = this.view.state.update({
            effects: StateEffect.appendConfig.of(EditorView.updateListener.of((v: ViewUpdate) => {
                if (v.docChanged) {
                    console.log(language.parse(v.state.doc.toString()).toString());
                }
            }))
        })
        this.view.dispatch(transaction);
```

Reply:
Thanks a lot for the explanation and the hint. It definitely saved me a lot of trial and error!